### ISSUE TITLE
Build failed under Ubuntu 18.04

### ISSUE THREAD

**Issue by ilyaevseev** [KostyaEsmukov/smtp_to_telegram#1]
### Install Golang:

```
$ sudo apt install -y golang-go
...
golang-1.10-go golang-1.10-race-detector-runtime golang-1.10-src golang-race-detector-runtime golang-src
...
```

### Try to install smtp2telegram:

```
$ go get github.com/KostyaEsmukov/smtp_to_telegram
# github.com/KostyaEsmukov/smtp_to_telegram
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:74:17: cannot use cli.StringFlag literal (type cli.StringFlag) as type cli.Flag in array or slice literal:
	cli.StringFlag does not implement cli.Flag (Apply method has pointer receiver)
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:78:10: unknown field 'EnvVar' in struct literal of type cli.StringFlag
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:80:17: cannot use cli.StringFlag literal (type cli.StringFlag) as type cli.Flag in array or slice literal:
	cli.StringFlag does not implement cli.Flag (Apply method has pointer receiver)
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:84:10: unknown field 'EnvVar' in struct literal of type cli.StringFlag
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:86:17: cannot use cli.StringFlag literal (type cli.StringFlag) as type cli.Flag in array or slice literal:
	cli.StringFlag does not implement cli.Flag (Apply method has pointer receiver)
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:89:10: unknown field 'EnvVar' in struct literal of type cli.StringFlag
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:91:17: cannot use cli.StringFlag literal (type cli.StringFlag) as type cli.Flag in array or slice literal:
	cli.StringFlag does not implement cli.Flag (Apply method has pointer receiver)
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:94:10: unknown field 'EnvVar' in struct literal of type cli.StringFlag
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:96:17: cannot use cli.StringFlag literal (type cli.StringFlag) as type cli.Flag in array or slice literal:
	cli.StringFlag does not implement cli.Flag (Apply method has pointer receiver)
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:100:10: unknown field 'EnvVar' in struct literal of type cli.StringFlag
go/src/github.com/KostyaEsmukov/smtp_to_telegram/smtp_to_telegram.go:96:17: too many errors
```

**Comment by KostyaEsmukov** [comment]
That probably means that some third-party dependency (probably https://github.com/urfave/cli) has been updated.

This project uses `go mod` which pins specific versions of dependencies. Unfortunately ubuntu 18.04 ships an older go version 1.10 (see https://packages.ubuntu.com/bionic/golang), whereas `go mod` is supported only since 1.12.

So under ubuntu 19.10 (which ships go 1.12) I have just built this project with no issues:

```
docker run -it --rm ubuntu:19.10 bash
$ apt update -qq
$ apt install -y -qq golang git
$ cd /root
$ git clone https://github.com/KostyaEsmukov/smtp_to_telegram.git
$ cd smtp_to_telegram/
$ go build
$ ./smtp_to_telegram -help
NAME:
   smtp_to_telegram - A small program which listens for SMTP and sends all incoming Email messages to Telegram.

USAGE:
   smtp_to_telegram [global options] command [command options] [arguments...]

VERSION:
   UNKNOWN_RELEASE

COMMANDS:
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --smtp-listen value          SMTP: TCP address to listen to (default: "127.0.0.1:2525") [$ST_SMTP_LISTEN]
   --smtp-primary-host value    SMTP: primary host (default: "61af4bc05e99") [$ST_SMTP_PRIMARY_HOST]
   --telegram-chat-ids value    Telegram: comma-separated list of chat ids [$ST_TELEGRAM_CHAT_IDS]
   --telegram-bot-token value   Telegram: bot token [$ST_TELEGRAM_BOT_TOKEN]
   --telegram-api-prefix value  Telegram: API url prefix (default: "https://api.telegram.org/") [$ST_TELEGRAM_API_PREFIX]
   --message-template value     Telegram message template (default: "From: {from}\\nTo: {to}\\nSubject: {subject}\\n\\n{body}") [$ST_TELEGRAM_MESSAGE_TEMPLATE]
   --help, -h                   show help
   --version, -v                print the version
```

There's also a Dockerfile which builds using `go mod` as well, so I'd suggest to just use the docker image. https://github.com/KostyaEsmukov/smtp_to_telegram/blob/master/Dockerfile

**Comment by KostyaEsmukov** [comment]
I have upgraded urfave/cli to v2 (c2c2e491f15bc534c759040227a3cf37e0ced8f1) which incidentally fixed `go get` on ubuntu 18.04:

```
docker run -it --rm ubuntu:18.04 bash
$ apt update -qq
$ apt install -y -qq golang git
$ cd
$ go get -v github.com/KostyaEsmukov/smtp_to_telegram
$ ~/go/bin/smtp_to_telegram -help
NAME:
   smtp_to_telegram - A small program which listens for SMTP and sends all incoming Email messages to Telegram.

USAGE:
   smtp_to_telegram [global options] command [command options] [arguments...]

VERSION:
   UNKNOWN_RELEASE

COMMANDS:
   help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --smtp-listen value          SMTP: TCP address to listen to (default: "127.0.0.1:2525") [$ST_SMTP_LISTEN]
   --smtp-primary-host value    SMTP: primary host (default: "f70a80450196") [$ST_SMTP_PRIMARY_HOST]
   --telegram-chat-ids value    Telegram: comma-separated list of chat ids [$ST_TELEGRAM_CHAT_IDS]
   --telegram-bot-token value   Telegram: bot token [$ST_TELEGRAM_BOT_TOKEN]
   --telegram-api-prefix value  Telegram: API url prefix (default: "https://api.telegram.org/") [$ST_TELEGRAM_API_PREFIX]
   --message-template value     Telegram message template (default: "From: {from}\\nTo: {to}\\nSubject: {subject}\\n\\n{body}") [$ST_TELEGRAM_MESSAGE_TEMPLATE]
   --help, -h                   show help (default: false)
   --version, -v                print the version (default: false)
```

But I would still suggest to use `go mod` or docker which would ensure that installation won't break again in future because of new versions in dependencies.
